pkg/sql/schemachanger/testdata/alter_table_add_column

```diff
@@ -2664,3 +2664,1454 @@ update progress of schema change job #1: "all stages completed"
+upsert descriptor #106
+  ...
+       - PUBLIC
+       - PUBLIC
+  -    - BACKFILL_ONLY
+  +    - DELETE_ONLY
+       - ABSENT
+       - WRITE_ONLY
+  ...
+     formatVersion: 3
+     id: 106
+  -  modificationTime:
+  -    wallTime: "1640995200000000003"
+  +  modificationTime: {}
+     mutations:
+     - column:
+  ...
+         version: 4
+       mutationId: 1
+  -    state: BACKFILLING
+  +    state: DELETE_ONLY
+     - direction: ADD
+       index:
+  ...
+       time: {}
+     unexposedParentSchemaId: 105
+  -  version: "24"
+  +  version: "25"
+upsert descriptor #107
+  ...
+     formatVersion: 3
+     id: 107
+  -  modificationTime:
+  -    wallTime: "1640995200000000003"
+  +  modificationTime: {}
+     name: sq1
+     parentId: 104
+  ...
+       start: "1"
+     unexposedParentSchemaId: 105
+  -  version: "10"
+  +  version: "11"
+update progress of schema change job #1: "PostCommitPhase stage 4 of 13 with 1 MutationType op pending"
+commit transaction #5
+begin transaction #6
+## PostCommitPhase stage 4 of 13 with 4 MutationType ops
+upsert descriptor #106
+  ...
+       - PUBLIC
+       - PUBLIC
+  -    - DELETE_ONLY
+  +    - MERGE_ONLY
+       - ABSENT
+       - WRITE_ONLY
+  ...
+     formatVersion: 3
+     id: 106
+  -  modificationTime:
+  -    wallTime: "1640995200000000005"
+  +  modificationTime: {}
+     mutations:
+     - column:
+  ...
+         version: 4
+       mutationId: 1
+  -    state: DELETE_ONLY
+  +    state: DELETE_AND_WRITE_ONLY
+     - direction: ADD
+       index:
+  ...
+       time: {}
+     unexposedParentSchemaId: 105
+  -  version: "25"
+  +  version: "26"
+upsert descriptor #107
+  ...
+     formatVersion: 3
+     id: 107
+  -  modificationTime:
+  -    wallTime: "1640995200000000005"
+  +  modificationTime: {}
+     name: sq1
+     parentId: 104
+  ...
+       start: "1"
+     unexposedParentSchemaId: 105
+  -  version: "11"
+  +  version: "12"
+update progress of schema change job #1: "PostCommitPhase stage 5 of 13 with 1 BackfillType op pending"
+commit transaction #6
+begin transaction #7
+## PostCommitPhase stage 5 of 13 with 1 BackfillType op
+merge temporary indexes [9] into backfilled indexes [8] in table #106
+commit transaction #7
+begin transaction #8
+## PostCommitPhase stage 6 of 13 with 1 ValidationType op
+validate forward indexes [8] in table #106
+commit transaction #8
+begin transaction #9
+## PostCommitPhase stage 7 of 13 with 10 MutationType ops
+upsert descriptor #106
+  ...
+       usesSequenceIds:
+       - 107
+  +  - defaultExpr: nextval(107:::REGCLASS)
+  +    id: 5
+  +    name: m
+  +    pgAttributeNum: 5
+  +    type:
+  +      family: IntFamily
+  +      oid: 20
+  +      width: 64
+  +    usesSequenceIds:
+  +    - 107
+     createAsOfTime:
+       wallTime: "1640995200000000000"
+  ...
+         userName: root
+       currentStatuses:
+  +    - VALIDATED
+  +    - ABSENT
+       - PUBLIC
+       - PUBLIC
+  -    - WRITE_ONLY
```

Comment:
I think this column is becoming public too early. It shouldn't become public until the index has been fully added. We don't want the logical side-effects of the schema change to leak if we're going to roll back. If you don't handle that with new rules here, please file an issue.